server/services/DeckService.js

```diff
@@ -103,12 +127,12 @@ class DeckService {
     async update(deck) {
         let previousVersion = await this.getById(deck.id);
         //do not save the deck if the deck is locked
-        if(previousVersion.locked) {
+        if(previousVersion.lockedForEditing) {
             throw new Error('Locked decks can not be updated');
         }
 
-        //if the eventId is set on the deck, check if the user already has a deck with the same eventId
-        if(deck.eventId) {
+        //if the eventId is set on the deck and the deck is not a draft deck, check if the user already has a deck with the same eventId
+        if(deck.eventId && deck.format !== 'draft') {
```

Comment:
Just musing aloud - I wonder if we should do any additional validation when "updating" draft decks. The `draftedCards` key isn't sent / updated in Mongo, so there's no potential cheating there (i.e. adding cards to the list you originally drafted). The deck validations from what I recall checks that only cards in the `draftedCards` list can appear in the other keys. Someone _could_ do a deck update with cards that don't appear in the `draftedCards` list and it would get saved; it would be marked as invalid, though.

Reply:
Agreed that a validation like this is necessary, but i´ve looked at the deck-helper code again and as you mentioned, it is already validated. Copying the code from the deck-helper doesn´t add much value i think. We could call the validate method from the deck-helper before updating though to make sure you can only save valid decks. What do you think?